### PR TITLE
Emptymask

### PR DESCRIPTION
**Pull request by jg10545**
Changes in this PR:

* Added keyword argument to `Trainer` that allows you to fit the mask generator without inpainting (closer to how the classifier is trained)
* Added keyword argument to `Trainer` that allows you to fit the inpainter with samples from the mask prior instead of the mask generator (the new mask prior seems pretty good- closing #32)
  * These two together are intended to suppress any weird interaction between inpainter and mask generator training. Seems to work so far.
* Added kwarg to `Trainer` for learning rate decay
* Added convenience code to `Trainer` for running inference on images (#20)
* Added more functionality to `Trainer` for fitting only the mask generator, only the inpainter, or different combinations of both
* Lots of under-the-hood cleanup because my code was getting out of hand.
* Better-optimized loading functions (shuffling filenames rather than entire batches of loaded images)

The main problem I'm running into right now is that saved inpainter models, reloaded with `tf.keras.models.load_model()` are giving different answers than the original model. Looks like there are a number of bugs with saving/loading nested models; I should probably update this for TF2.0 before going too far down any of those rabbit holes.

### CODE REVIEW

**Comment by jg10545** [comment]
I'm going to go ahead and merge this- let me know if you've got any questions/concerns @zamorarr
